kong/plugins/http-log/handler.lua

```diff
@@ -189,12 +190,27 @@ function HttpLogHandler:log(conf)
 
   local queue_conf = Queue.get_plugin_params("http-log", conf, make_queue_name(conf))
   kong.log.debug("Queue name automatically configured based on configuration parameters to: ", queue_conf.name)
+  local entry = cjson.encode(kong.log.serialize())
+
+  if queue_conf.max_batch_size == 1 then
+    local queue = Queue.create(queue_conf, send_entries, conf)
+    local ok, err = timer_at(0, function(premature)
+      if premature then
+        return
+      end
+      queue:handle({ entry })
```

Comment:
the problem with this approach is that we'd be using the Queue to process entries in parallel, i.e. not what a queue normally does. To properly reuse the code (now in the queue) that processes an entry we should probably extract it from the Queue module and then refer to it both from the queue and from here.

If this is too complex for the scope of this task we might want to do it as a separate refactor.

Reply:
I agree that we extract the retry logic ourside of Queue in another PR.